.gitlab/trigger_release/installer.yml

```
@@ -0,0 +1,73 @@
+---
+# trigger_release stage
+# Contains jobs related to triggering the release of the Datadog Installer.
+
+# TODO: This is currently separate from the base agent release management trigger as we want to ensure the Agent is not impacted
+# as we adopt the `dd-pkg` tooling for the Installer. We should deduplicate the code between the two files in the future.
+.installer_release_management_trigger:
+  stage: trigger_release
+  image: registry.ddbuild.io/ci/datadog-agent-buildimages/deb_x64$DATADOG_AGENT_BUILDIMAGES_SUFFIX:$DATADOG_AGENT_BUILDIMAGES
+  tags: ["arch:amd64"]
+  variables:
+    ACTION: promote
+    BUILD_PIPELINE_ID: $CI_PIPELINE_ID
+    RELEASE_PRODUCT: datadog-installer
+    RELEASE_VERSION: $CI_COMMIT_TAG
+    TARGET_REPO_BRANCH: $BUCKET_BRANCH
+  script:
+    - GITLAB_TOKEN=$($CI_PROJECT_DIR/tools/ci/fetch_secret.sh $GITLAB_TOKEN write_api) || exit $?; export GITLAB_TOKEN
+    - 'dda inv pipeline.trigger-child-pipeline --project-name "DataDog/agent-release-management" --git-ref "main"
+      --variable ACTION
+      --variable AUTO_RELEASE
+      --variable BUILD_PIPELINE_ID
+      --variable RELEASE_PRODUCT
+      --variable RELEASE_VERSION
+      --variable TARGET_REPO
+      --variable TARGET_REPO_BRANCH
+      $NO_FOLLOW'
+  dependencies: []
+
+installer_trigger_auto_staging_release:
+  extends: .installer_release_management_trigger
+  timeout: 2h 30m
+  variables:
+    AUTO_RELEASE: "true"
+    TARGET_REPO: staging
+  rules:
+    - if: $CI_COMMIT_TAG =~ /^[0-9]+\.[0-9]+\.[0-9]+-installer-[0-9]+\.[0-9]+\.[0-9]+(-rc\.[0-9]+){0,1}$/
+      when: never
+    - !reference [.on_deploy]
```

It's best to deploy the installer when the user uses the `--deploy-installer` option.
Maybe we should create other references for the other jobs, too

```suggestion
    - !reference [.on_deploy_installer]
```

---

That is to say, you still want to have the whole pipeline triggered manually locally via the invoke task? That should be fine, given the triggering is happening in pipeline still.

---

A bunch of jobs / templates such as `.deploy_installer_deb` are only ran with the rules in `.on_deploy_installer`.
Though tbh I'm not against using the `CI_COMMIT_TAG` condition in .`on_deploy_installer`

---

`CI_COMMIT_TAG` isn't going to work with the existing workflow, it won't be present on the pipeline y'all trigger with the invoke task - but it's something we could reintroduce in the future, if we wanted to configure the jobs to run on just tags and not from the invoke task.

---

Got a successful run here.
https://gitlab.ddbuild.io/DataDog/agent-release-management/-/pipelines/61664587

the upstream pipeline triggered with:
```
dda inv -e pipeline.run --deploy-installer --git-ref="7.66.0-installer-release-trigger" --repo-branch="beta"
```